src/backend/distributed/ddl/foreign_constraint.c

```diff
@@ -309,6 +369,94 @@ ErrorIfUnsupportedForeignConstraint(Relation relation, char distributionMethod,
 }
 
 
+bool
+ForeignKeyExistsToReferenceTableOnColumn(char *affectedColumnName,
```

Comment:
1) Missing function comment.
2) maybe `affectedColumnName ` - > `columnName`. 
3) name seems to fit to a single line, so no need to have a new line?

Maybe more importantly, the name seems like not covering all cases, because the function returns true if it finds the key in the reference table as well? Seems like ` `ColumnAppearInForeignKey()` or such more generic names fit better